epictrack-web/src/components/insights/Work/Tabs/Staff/Charts/workListing.tsx

```diff
@@ -68,6 +71,103 @@ const WorkList = () => {
     );
   }, [workStaffs]);
 
+  const filteredStaffByPosition = useCallback(
+    (position: WorkStaffRole) => {
+      if (!workStaffs) return [];
+      const staff = workStaffs.flatMap((row: any) =>
+        row.staff
+          ? row.staff.filter((p: { role: Role }) => p.role.id === position)
+          : []
+      );
+      const staffSorted = sort(staff, "full_name");
+      const uniqueStaffNames = Array.from(
+        new Set(
+          staffSorted.map(
+            (staffEntry: any) =>
+              `${staffEntry.last_name}, ${staffEntry.first_name}`
+          )
+        )
+      );
+      return uniqueStaffNames;
+    },
+    [workStaffs]
+  );
+
+  const coLeadOptions = filteredStaffByPosition(WorkStaffRole.TEAM_CO_LEAD);
+  const officerAnalystOptions = filteredStaffByPosition(
+    WorkStaffRole.OFFICER_ANALYST
+  );
+
+  const roleFilterFunction = (row: any, id: any, filterValue: any) => {
+    const options =
+      id === WorkStaffRoleNames[WorkStaffRole.OFFICER_ANALYST]
+        ? officerAnalystOptions
+        : coLeadOptions;
+    if (
+      !filterValue.length ||
+      filterValue.length > options.length // select all is selected
+    ) {
+      return true;
+    }
+
+    const value: string = row.getValue(id) || "";
+    // Split the cell value into individual names
+    const names = value.split("; ");
+
+    // Check if any name includes the filter value
+    return names.some((name) => filterValue.includes(name));
+  };
+
+  const getRolfilterOptions = (role: WorkStaffRole) => {
+    return role === WorkStaffRole.OFFICER_ANALYST
+      ? officerAnalystOptions
+      : coLeadOptions;
+  };
+
+  const tableColumns = React.useMemo(() => {
```

Comment:
What is this used for? Where?

Reply:
the insights was not displaying other roles like officer analyst or team colead